### PR TITLE
fix: `res/app/job` consumption plan usage

### LINKED ISSUE #2517: [AVM Module Issue]: Deploying jobs to "Consumption only" Container Apps environment fails
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/app/job

### (Optional) Module Version

0.2.1

### Description

In case the Container Apps Environment is "Consumption Only" (i.e. now workloadprofile has been defined), then even the bare minimum [default job](https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/app/job/README.md#example-1-using-only-defaults) deployment script fails with the following message:

`Cannot specify workload profile name for container app in Consumption Only environment.` 

`workloadProfileName` currently defaults to `Consumption` int the bicep module, but I think the default should be `null`

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by ReneHezser**
## Description

Clarify and fix usage of consumption plan.

Fixes #2517 

## Pipeline Reference

| Pipeline |
| -------- |
|   [avm.res.app.job](https://github.com/ReneHezser/bicep-registry-modules/actions/workflows/avm.res.app.job.yml)       |

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [x] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [x] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [x] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/app/job/main.bicep [hunk 1]
```diff
@@ -86,8 +86,8 @@ param manualTriggerConfig manualTriggerConfigType?
 @description('Optional. The maximum number of times a replica can be retried.')
 param replicaRetryLimit int = 0
 
-@description('Optional. The name of the workload profile to use.')
-param workloadProfileName string = 'Consumption'
+@description('Optional. The name of the workload profile to use. Leave empty to use a consumption based profile.')
```
Comment (AlexanderSehr) [inline avm/res/app/job/main.bicep:89]: Hey @ReneHezser, really 'empty' (aka an empty string), or should the user then not specify a value? Or does both lead to 'Consumption'?

File: avm/res/app/job/main.bicep [hunk 2]
```diff
@@ -114,6 +115,7 @@ param workloadProfileName string = 'Consumption'
   ]
   '''
 })
+#disable-next-line secure-secrets-in-params // not a secret, just a reference to a secret
```
Comment (AlexanderSehr) [inline avm/res/app/job/main.bicep:118]:
```suggestion
#disable-next-line secure-secrets-in-params // @secure() is specified in UDT
```

File: avm/res/app/job/tests/e2e/consumptionPlan/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,66 @@
+targetScope = 'subscription'
+
+metadata name = 'Using a consumption plan'
```
Comment (AlexanderSehr) [inline avm/res/app/job/tests/e2e/consumptionPlan/main.test.bicep:3]: General question - is this test case not already covered by the min test which, I assume, defaults to a 'consumption' plan?

File: avm/res/app/job/main.bicep (hunk 1 above)
Comment (peterbud) [reply avm/res/app/job/main.bicep:89]: If I may add here (I have worked with Rene on this PR, and I have submitted the original issue): It should be empty, as in the PR.

There are two types of Container App Environments:
* Consumption only environment: if your `workloadProfileName` is empty, a "consumption only" type of environment will be created
* Workload profiles environment: if you specify a non-empty `workloadprofileName`, a "workload profiles" type of environment will be created. Even if you workloadprofile nami is "Consumption"- that would be still considered as "Workload profiles" type environment.

More [here](https://learn.microsoft.com/en-us/azure/container-apps/environment)

File: avm/res/app/job/tests/e2e/consumptionPlan/main.test.bicep (hunk 3 above)
Comment (AlexanderSehr) [reply avm/res/app/job/tests/e2e/consumptionPlan/main.test.bicep:3]: Yes, no, maybe @ReneHezser ? 😄

Comment (ReneHezser) [reply avm/res/app/job/tests/e2e/consumptionPlan/main.test.bicep:3]: The default test is not using the consumption plan.
image

The consumption plan test ist using it.
image

I created the additional test to test this.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**